src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -2309,8 +2416,17 @@ void ApplicationManagerImpl::ProcessMessageFromHMI(
   *smart_object = message->smart_object();
 #else
   if (!ConvertMessageToSO(*message, *smart_object)) {
-    LOG4CXX_ERROR(logger_, "Cannot create smart object from message");
-    return;
+    if (application_manager::MessageType::kResponse ==
+        (*smart_object)[strings::params][strings::message_type].asInt()) {
+      (*smart_object).erase(strings::msg_params);
+      (*smart_object)[strings::params][hmi_response::code] =
+          hmi_apis::Common_Result::INVALID_DATA;
```

Comment:
Honestly, we shouldn't be sending an `INVALID_DATA` to the mobile proxy in this case, since it refers to invalid data from the HMI. Tt seems to me like we should be something like `ABORTED` or `GENERIC_ERROR`, since the mobile device can't really do anything to fix the issue.

Reply:
@jacobkeeler some RC use cases depends from  this staff.